<header class="banner"><div class="banner">
  <div class="logos">
    <div><img title="Julia logo" src="figures/julia-logo-color.svg" alt="Julia animated logo" width=35 /></div>
    <div>◌</div>
    <div><img title="Powered by MathJax" src="https://www.mathjax.org/badge/mj-logo.svg"
              alt="Powered by MathJax" width=65 />
    </div>
  </div>  <!-- logos -->
  <div class="cite_textbook cb8">3.029<span class="authors"></span></div>
</div></header>





### Given 
A piston–cylinder device contains 0.85&nbsp;kg of R-134a at −10&nbsp;°C. The piston, which is free to move, has a mass of 12&nbsp;kg and a diameter of 25&nbsp;cm. The local atmospheric pressure is 88&nbsp;kPa. Now, heat is transferred to R-134a until the temperature is 15&nbsp;°C. 

### Find
1. the final pressure,  
1. the change in the volume of the cylinder, and  
1. the change in the enthalpy of the R-134a.

### Get Organized

A good diagram goes a long way. Draw it and USE IT. 

![cb8,3-29](figures/cb8,3-29.png)

Our system is inside a piston-cylinder device. It's volume can change; the system is screaming to you **moving boundary work**. We aren't asked to find work, but let it scream regardless. 

The piston is free to move: now the system is screaming **constant pressure (isobaric) process**.

Now let's look at what we were asked to find:  

1. Final Pressure.  
 We have an isobaric process, so \begin{gather} \notag \\ P_{\rm 2} = P_{\rm 1}. \end{gather}  
 We are not given initial pressure $P_{\rm1},$ but we can calculate it: 
 \begin{gather}  \notag \\
  P_{\rm 1} = \frac{F_{\rm sys}}{A_{\rm sys}}. \\ \notag{}
 \end{gather} 
 Of course, the weight of the piston acts on our system. Working with absolute pressure is far more robust than working with gage pressure; that requires us to account for the force applied by atmospheric pressure.  
  \begin{gather}
   \begin{aligned} \notag \\
  P_{\rm 1} &= \frac{F_\text{atm} + F_\text{piston} }{A} \\[9pt]
            &= \frac{P_\text{atm}A + m_\text{piston}g }{A} \\[9pt] 
            &= P_\text{atm} + \frac{m_\text{piston}g }{A}  
 \end{aligned}
 \end{gather}

1. Volume change.  

 We seek  
 \begin{gather}  
   \Delta{V} = V_2 - V_1.\\ \notag{}
 \end{gather} 
 We know that specific volume relates volume and mass via  
 \begin{gather} \notag \\
  V = mv. \\ \notag{}
 \end{gather}
  Thus, 
  \begin{equation}
  \begin{gathered}  \notag \\
   \Delta{V} = V_2 - V_1 = mv_2 - mv_1 \\[9pt]
   \Delta{V} = m \left(v_2 - v_1 \right) \\ \notag{}
 \end{gathered} 
 \end{equation}  
 
 $$ {} $$
  $$\bf \text {We need }v_1, v_2 $$
 $$ {} $$
 
1. Enthalpy change.  

 We seek  
 \begin{gather}  
   \Delta{H} = H_2 - H_1.\\ \notag{}
 \end{gather} 
 We know that specific enthalpy relates enthalpy and mass via  
 \begin{gather} \notag \\
  H = mh. \\ \notag{}
 \end{gather}
  Thus, 
  \begin{equation}
  \begin{gathered}  \notag \\
   \Delta{H} = H_2 - H_1 = mh_2 - mh_1 \\[9pt]
   \Delta{H} = m \left(h_2 - h_1 \right) \\ \notag{}
 \end{gathered} 
 \end{equation} 
  $$\bf \text {We need }h_1, h_2 $$

### Attack

Let's get to work. 

In [1]:
# ⋮⋮⋮⋮⋮ Julia initializations and setups ⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮⋮
using Unitful     
no_units = u"m"/u"m"

g        =  1u"gn"    # Unitful quantity, sea level Earth gravity
P_atm    = 88u"kPa"   # given  
D_piston = 25u"cm"    # given 
m_piston = 12u"kg"    # given 
m_sys    = 0.85u"kg"  # given

A = π/4 * D_piston^2
@show A;

A = 490.8738521234052 cm^2


#### 1. Final pressure P₂ 


In [2]:
P₁ = P_atm + m_piston*g/A     # eqn(3)
P₂ = P₁ |> u"kPa"             # eqn(1) 
@show P₂

P₂ = 90.3973531996244 kPa


90.3973531996244 kPa

<div class="alert alert-success" role="alert" style="border: 1px solid #c3e6cb;">

$$ P_2 = \pu{90.40 kPa = 0.9040 bar} $$

</div>

#### Properties

We need $v_1, v_2, h_1,h_2.$  
$\begin{gather*} \notag \\
 \quad v_1: \left\{ 
 \begin{split} 
  T_1 &= \ce{-10°C} \\
  P_1 &= 0.904\, \ce{bar}
\end{split} \right| P_\text{sat@-10°C} = 2.0122\, \ce{bar}.\quad P_1 < P_\ce{sat}. \\ \notag{}
 \end{gather*} \quad \therefore$ State 1 is superheated vapor.  
 $P_1$ is not a breakpoint in the R-134a Superheated Vapor table, so we must interpolate. We know we need $h_1$, so we'll get it while we're getting $v$. The process is the same; only the table columns are different.  
 
 \begin{align*}
 \quad  v_\text{-10°C, 0.6 bar} &= 0.34992\,\ce{m^3/kg}  & h_\text{-10°C, 0.6 bar} &= 245.96\,\ce{kJ/kg} \\
 \quad  v_\text{-10°C, 1.0 bar} &= 0.20686\,\ce{m^3/kg}  & h_\text{-10°C, 1.0 bar} &= 244.70\,\ce{kJ/kg} \\
 \end{align*}
 
 $$\begin{array}{ccc}
   P_1  &-& 0.6\,  \ce{bar} \\
  \hline
  1.0\,\ce{bar} &-& 0.6\,\ce{bar}
 \end{array}
 = 
 \begin{array}{ccc}
  v_1  &-& v_\text{-10°C, 0.6 bar} \\
  \hline
  v_\text{-10°C, 1.0 bar} &-& v_\text{-10°C, 0.6 bar}
 \end{array}
 = 
 \begin{array}{ccc}
  h_1  &-& h_\text{-10°C, 0.6 bar} \\
  \hline
  h_\text{-10°C, 1.0 bar} &-& h_\text{-10°C, 0.6 bar}
 \end{array}$$ 


In [3]:
v_⁻10°C_60kPa  = 0.34992u"m^3/kg"          # Julia won't allow a 
v_⁻10°C_100kPa = 0.20686u"m^3/kg"          # decimal point in a variable name, hence kPa

h_⁻10°C_60kPa  = 245.96u"kJ/kg"
h_⁻10°C_100kPa = 244.70u"kJ/kg"

# interpolation_factor makes it neater when interpolating multiple properties at once
interpolation_factor = (P₁ - 0.6u"bar") / (1.0 - 0.6)u"bar" |> no_units  
@show interpolation_factor; 

interpolation_factor = 0.75993382999061


In [4]:
v₁ = v_⁻10°C_60kPa + interpolation_factor * (v_⁻10°C_100kPa - v_⁻10°C_60kPa) 
h₁ = h_⁻10°C_60kPa + interpolation_factor * (h_⁻10°C_100kPa - h_⁻10°C_60kPa) 

@show v₁ h₁;

v₁ = 0.24120386628154333 m^3 kg^-1
h₁ = 245.00248337421183 kJ kg^-1


<p style = "background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

We've got State 1. Now we resolve State 2.

$\begin{gather*} \notag \\
 \quad v_2: \left\{ 
 \begin{split} 
  T_2 &= \ce{15°C} \\
  P_2 &= 0.904\, \ce{bar}
\end{split} \right| P_\text{sat@15°C} = \ ? \\ \notag{}
 \end{gather*} $   
 Neither $P_2$ nor $T_2$ are breakpoints in the R-134a Saturation or Superheated Vapor tables, so we must interpolate a bit more. First, we need to determine our phase at State 2, so we head to the Saturation Tables to get $P_\ce{sat}$ and compare it to $P_2$. From the Saturation Temperature Table: 
 
 | $T$ [°C] | $P_\ce{sat}$ [bar] |
 |:--------:|:------------------:|
 |    12    |       4.4294       |
 |    16    |       5.0416       |
 
 We could interpolate to find $P_\ce{sat}$, but we don't need the actual value of the saturation pressure. To wit: we know $P_\text{sat@15°C}$ > 4.4294 bar, which means $P_2 = 0.904\,\ce{bar}\lt P_\text{sat@15°C}. \quad \therefore$ State 2 is superheated vapor.  
 
 

We have to perform 2-dimensional interpolation to resolve State 2. It doesn't matter whether you do temperature or pressure first. Here's a little table compiled from the Superheated Vapor table for the pressures and temperatures that bracket our state $\big( P_2 = 0.904\,\ce{bar}, T_2 = 15 \ce{°C} \big)$.
 
 
 | $P$ [bar] | $T$ [°C] | $v$ [m³/kg] | $h$ [kJ/kg] |
 |:---------:|:--------:|:-----------:|:-----------:|
 |    0.6    |    10    |   0.37861   |    262.41   |
 |    0.6    |    20    |   0.39279   |    270.89   |
 |    1.0    |    10    |   0.22473   |    261.43   |
 |    1.0    |    20    |   0.23349   |    270.02   |
 
 


 _Temperature_  

$T_2$ is halfway between breakpoints, so the interpolation reduces to the average of the breakpoint values:
 
 \begin{align*}
  v_\text{15°C, 0.6 bar} &= \textstyle \frac12 \big( v_\text{10°C, 0.6 bar} + v_\text{20°C, 0.6 bar} \big) & 
  h_\text{15°C, 0.6 bar} &= \textstyle \frac12 \big( h_\text{10°C, 0.6 bar} + h_\text{20°C, 0.6 bar} \big) \\[6pt]
  v_\text{15°C, 1.0 bar} &= \textstyle \frac12 \big( v_\text{10°C, 1.0 bar} + v_\text{20°C, 1.0 bar} \big) & 
  h_\text{15°C, 1.0 bar} &= \textstyle \frac12 \big( h_\text{10°C, 1.0 bar} + h_\text{20°C, 1.0 bar} \big) \\{}
 \end{align*}
 
Resulting in the additions rendered in bold red:

 | $\bf P$ [bar] | $\bf T$ [°C] | $\bf v$ [m³/kg] | $\bf h$ [kJ/kg]  |
 |:-------------:|:------------:|:---------------:|:----------------:|
 |      0.6      |     10       |    0.37861      |      262.41      |
 | $${\color{red} 0.6}$$ | $${\color{red} 15}$$  | $${\color{red}0.38570}$$ | ${\color{red}266.650}$ |
 |      0.6      |     20       |   0.39279       |       270.89      |
 |      1.0      |     10       |   0.22473       |       261.43      |
 | $$\color{red}1.0$$ | $${\color{red}15}$$ | $${\color{red}0.22911}$$ | $${\color{red}265.725}$$ |
 |      1.0      |     20       |   0.23349       |       270.02      |

 

 _Pressure_ 
 
 Now, we interpolate over pressure and wrap up.

 $$\begin{array}{ccc}
   P_2  &-& 0.6\,  \ce{bar} \\
  \hline
  1.0\,\ce{bar} &-& 0.6\,\ce{bar}
 \end{array}
 = 
 \begin{array}{ccc}
  v_2  &-& v_\text{15°C, 0.6 bar} \\
  \hline
  v_\text{15°C, 1.0 bar} &-& v_\text{15°C, 0.6 bar}
 \end{array}
 = 
 \begin{array}{ccc}
  h_2  &-& h_\text{15°C, 0.6 bar} \\
  \hline
  h_\text{15°C, 1.0 bar} &-& h_\text{15°C, 0.6 bar}
 \end{array}$$ 

In [5]:
# Julia won't allow a decimal point in a variable name, hence kPa
v_15°C_60kPa  = 0.38570u"m^3/kg" 
v_15°C_100kPa = 0.22911u"m^3/kg" 

h_15°C_60kPa  = 266.650u"kJ/kg"
h_15°C_100kPa = 265.725u"kJ/kg"

# interpolation_factor makes it neater when interpolating multiple properties at once
interpolation_factor = (P₁ - 0.6u"bar") / (1.0 - 0.6)u"bar" |> no_units  
@show interpolation_factor;

interpolation_factor = 0.75993382999061


In [6]:
v₂ = v_15°C_60kPa + interpolation_factor * (v_15°C_100kPa - v_15°C_60kPa) 
h₂ = h_15°C_60kPa + interpolation_factor * (h_15°C_100kPa - h_15°C_60kPa) 
@show v₂ h₂;

v₂ = 0.2667019615617704 m^3 kg^-1
h₂ = 265.9470612072587 kJ kg^-1


#### 2. Change in the volume of the R-134a    

 $\Delta V = V_2 - V_1 = m\left( v_2 - v_1 \right)$

In [7]:
ΔV = m_sys * (v₂ - v₁)
@show ΔV;

ΔV = 0.021673380988193002 m^3


<div class="alert alert-success" role="alert" style="border: 1px solid #c3e6cb;">

$$ \pu{\Delta V = 0.0217 m^3} $$
  
</div>

#### 3. Change in the enthalpy of the R-134a

$$\Delta H = H_2 - H_1 = m\left( h_2 -h_1 \right)$$

In [8]:
ΔH = m_sys * (h₂ - h₁)
@show ΔH;

ΔH = 17.80289115808982 kJ


<div class="alert alert-success" role="alert" style="border: 1px solid #c3e6cb;">
  
$$\pu{\Delta H = 17.803 kJ}$$

</div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>